packages/esm-patient-orders-app/src/order-cancellation-form/cancel-order-form.component.tsx

```diff
@@ -0,0 +1,182 @@
+import { saveCancelOrderRequest } from './cancel-order.resource';
+import dayjs from 'dayjs';
+
+interface OrderCancellationFormProps {
+  order: Order;
+  patientUuid: string;
+  context?: string;
+  closeWorkspace: () => void;
+}
+
+const OrderCancellationForm: React.FC<OrderCancellationFormProps> = ({ order, patientUuid, closeWorkspace }) => {
+  const { t } = useTranslation();
+  const isTablet = useLayoutType() === 'tablet';
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [showErrorNotification, setShowErrorNotification] = useState(false);
+
+  const { mutate } = usePatientOrders(patientUuid, 'ACTIVE');
+
+  const cancelOrderSchema = useMemo(() => {
+    return z.object({
+      cancellationDate: z.date(),
+      reasonForCancellation: z.string(),
+    });
+  }, []);
+
+  type CancelOrderFormData = z.infer<typeof cancelOrderSchema>;
+
+  const { control, handleSubmit, watch, setValue } = useForm<CancelOrderFormData>({
+    mode: 'all',
+    resolver: zodResolver(cancelOrderSchema),
+  });
+
+  function onError(err) {
+    if (err?.oneFieldRequired) {
+      setShowErrorNotification(true);
+    }
+  }
+
+  const cancelOrderRequest = useCallback((data: CancelOrderFormData) => {
+    const formData = data;
+    setShowErrorNotification(false);
+    setIsSubmitting(true);
+
+    const payload = {
+      fulfillerStatus: 'DECLINED',
+      fulfillerComment: formData.reasonForCancellation,
+    };
+
+    const abortController = new AbortController();
+
+    saveCancelOrderRequest(order, payload, abortController).then(
+      (res) => {
+        setIsSubmitting(false);
+        closeWorkspace();
+        mutate();
+
+        showSnackbar({
+          isLowContrast: true,
+          title: t('cancelOrder', 'Cancel Order'),
+          kind: 'success',
+          subtitle: t(
+            'successfullyCancelledOrder',
+            'You have successfully cancelled an order with OrderNumber {{orderNumber}}',
+            { orderNumber: order.orderNumber },
+          ),
+        });
+      },
+      (err) => {
+        setIsSubmitting(false);
+        showSnackbar({
+          isLowContrast: true,
+          title: t(`errorCancellingOrder', 'Error cancelling order`),
+          kind: 'error',
+          subtitle: err?.message,
+        });
+      },
+    );
+  }, []);
+
+  return (
+    <Form className={styles.form}>
+      <div className={styles.grid}>
+        <Stack>
+          <section>
+            <h4 style={{ marginBottom: '1rem' }}>{order.display}</h4>
+          </section>
+          <section>
+            <Controller
+              name="cancellationDate"
+              control={control}
+              render={({ field: { onChange, value } }) => (
+                <div className={styles.row}>
+                  <DatePicker
+                    id="vaccinationExpiration"
+                    className="vaccinationExpiration"
+                    minDate={dayjs().startOf('day').format('DD/MM/YYYY')}
```

Comment:
@vasharma05 useful to check that this doesn't fall prey to the minDate issue you're tackling.

Reply:
I shared my review on this, we can directly pass `dayjs().startOf('day')` without formatting.